azurelinuxagent/ga/monitor.py

```diff
@@ -242,11 +248,13 @@ def collect_and_send_events(self):
     def daemon(self):
         min_delta = min(MonitorHandler.TELEMETRY_HEARTBEAT_PERIOD,
                         MonitorHandler.EVENT_COLLECTION_PERIOD,
-                        MonitorHandler.HOST_PLUGIN_HEARTBEAT_PERIOD).seconds
+                        MonitorHandler.HOST_PLUGIN_HEARTBEAT_PERIOD,
+                        MonitorHandler.IMDS_HEARTBEAT_PERIOD).seconds
         while self.should_run:
             self.send_telemetry_heartbeat()
             self.collect_and_send_events()
```

Comment:
Please open an issue to re-visit this.  I feel like we should be managing these individual things as a whole, or providing better support for them.

Reply:
Ok, but I'm not quite sure what you mean. Let's sync offline.

Reply:
Agreed - opened #1202